### DISCUSSION TITLE
whether the yalmip optimizer can reduce the yalmip time in the large scale problem?

### DISCUSSION THREAD

**Discussion by arielj-weibo** [yalmip/YALMIP#1106] [General discussion]
In my model, it is a linear optimization which uses the solver 'cplex' in matlab. 
The problem's loop is about 1000 times and there is only one parameter that changes each loop.
However, i found that the model using optimizer is slower than that doesn't use the optimizer.
But my goal is to speed up the model.
So my question is whether the yalmip optimizer is suitable for the larger scale than mine or the parameter has some other restrictions?
I really hope for your reply.

**Comment by johanlofberg** [comment]
Hard to answer without reproducible example. 

Most likely you have a very complicated parameterization, either just being very large, or complex such as involving stuff like x * function(parameters) which means you should look at that as x * newparameter and then send function(parametervalues) when calling the object

**Reply by arielj-weibo** [reply]
Thank you for your reply!

Although the model only has 1 parameter _x_ that needs to be parameterized, it is not a single number but a vector(50*1).

Because the constraint in one loop also varies with the parameter. Thus, when i want to use the optimizer, i need to write _x=sdpvar(100,1)_ not just _sdpvar x_

**Reply by johanlofberg** [reply]
You will have to supply a reproducible example. Having 50 parameters is not a problem per se, it depends on what the model looks like.

**Reply by johanlofberg** [reply]
Reproducible means the code runs and illustrates the issue.
